index.d.ts

```diff
@@ -1970,10 +1971,10 @@ declare module 'mongoose' {
       isMongooseArray: true;
 
       /** Pushes items to the array non-atomically. */
-      nonAtomicPush(...args: any[]): number;
+      nonAtomicPush(...args: T[]): number;
```

Comment:
I'd make this `AnyKeys<T> & AnyObject` instead of `T[]`. Mongoose handles casting, so you don't necessarily have to push an instance of `T`.

Reply:
Hello,
I have made the changes you requested